addOns/ascanrulesAlpha/src/main/java/org/zaproxy/zap/extension/ascanrulesAlpha/MongoDbInjectionTimingScanRule.java

```diff
@@ -0,0 +1,317 @@
+                    + "db.injection.mapReduce(function() { emit(1,1",
+            "_id);}, function(inj) { sleep("
+                    + INSERT_LONG_TIME
+                    + "); return 1;}, { out: 'x'}); "
+                    + "db.injection.mapReduce(function() { emit(1,1"
+        }
+    };
+    // Log prints
+    private static final String IO_EX_LOG = "trying to send an http message";
+    private static final String STOP_LOG = "Stopping the scan due to a user request";
+    private static final Logger LOGGER = LogManager.getLogger(MongoDbInjectionTimingScanRule.class);
+    private static final Map<String, String> ALERT_TAGS =
+            CommonAlertTag.toMap(
+                    CommonAlertTag.OWASP_2021_A03_INJECTION,
+                    CommonAlertTag.OWASP_2017_A01_INJECTION,
+                    CommonAlertTag.WSTG_V42_INPV_05_SQLI);
+    // Variables
+    private boolean doTimedScan;
+
+    @Override
+    public int getCweId() {
+        return 943;
+    }
+
+    @Override
+    public int getWascId() {
+        return 19;
+    }
+
+    @Override
+    public int getId() {
+        return 90039;
+    }
+
+    @Override
+    public boolean targets(TechSet technologies) {
+        return technologies.includes(Tech.MongoDB);
+    }
+
+    @Override
+    public String getName() {
+        return Constant.messages.getString(MESSAGE_PREFIX + "name");
+    }
+
+    @Override
+    public int getRisk() {
+        return Alert.RISK_HIGH;
+    }
+
+    @Override
+    public String getDescription() {
+        return Constant.messages.getString(MESSAGE_PREFIX + "desc");
+    }
+
+    @Override
+    public int getCategory() {
+        return Category.INJECTION;
+    }
+
+    @Override
+    public String getSolution() {
+        return Constant.messages.getString(MESSAGE_PREFIX + "soln");
+    }
+
+    @Override
+    public String getReference() {
+        return Constant.messages.getString(MESSAGE_PREFIX + "refs");
+    }
+
+    public String getExtraInfo(String attack) {
+        return Constant.messages.getString(MESSAGE_PREFIX + "extrainfo." + attack);
+    }
+
+    @Override
+    public Map<String, String> getAlertTags() {
+        return ALERT_TAGS;
+    }
+
+    @Override
+    public void init() {
+        LOGGER.debug("Initialising MongoDB penetration tests");
+        switch (this.getAttackStrength()) {
+            case LOW:
+                SLEEP_SHORT_TIME = 1000;
+                SLEEP_LONG_TIME = 2000;
+                doTimedScan = true;
+                break;
+            default:
+                SLEEP_SHORT_TIME = 1000;
+                SLEEP_LONG_TIME = 3000;
+                doTimedScan = true;
+                break;
+        }
+    }
+
+    @Override
+    public void scan(HttpMessage msg, NameValuePair originalParam) {
+        super.scan(msg, originalParam);
+    }
+
+    @Override
+    public void scan(HttpMessage msg, String param, String value) {
+        List<Integer> rtt = new ArrayList<>();
```

Comment:
This requires more changes, the time based attack expects that messages were sent to calculate the average RTT, currently this is leading to `ArithmeticException: / by zero`.